src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -139,6 +149,13 @@ message Resource {
     // - Executables: process_id, executable_path, working_directory, arguments
     // - Projects: process_id, project_path
     repeated ResourceProperty properties = 12;
+
+    repeated Url urls = 13;
+
+    // Returns true if there will ever be any urls. Since these updates are being sent as the state
+    // of resources change, it is possible to for urls to be empty but expect_urls to be true.
+    // If urls is non-empty, expect_urls must be true.
+    bool expect_urls = 14;
```

Comment:
A `Url` isn't present in the urls collection until it is started/ready, right? And that's why this property is required?

Is there a reason we couldn't instead have all the named urls in the urls collection no matter what, and then have a state on url that indicates whether it's ready?

Reply:
It’s not required. We can remove it, but the dashboard will go from None -> URLs instead of Starting -> URLs. I don’t think it’s worth it having this complexity. I was just trying to preserve the current state

Reply:
Removed